application/github/v0/main.go

```diff
@@ -0,0 +1,179 @@
+	taskGetAllIssues        = "TASK_GET_ALL_ISSUES"
+	taskGetIssue            = "TASK_GET_ISSUE"
+	taskCreateIssue         = "TASK_CREATE_ISSUE"
+	taskCreateWebhook       = "TASK_CREATE_WEBHOOK"
+)
+
+var (
+	//go:embed config/definition.json
+	definitionJSON []byte
+	//go:embed config/setup.json
+	setupJSON []byte
+	//go:embed config/tasks.json
+	tasksJSON []byte
+
+	once sync.Once
+	comp *component
+)
+
+type component struct {
+	base.Component
+}
+
+type execution struct {
+	base.ComponentExecution
+	execute func(*structpb.Struct) (*structpb.Struct, error)
+	client  Client
+}
+
+// Init returns an implementation of IConnector that interacts with Slack.
+func Init(bc base.Component) *component {
+	once.Do(func() {
+		comp = &component{Component: bc}
+		err := comp.LoadDefinition(definitionJSON, setupJSON, tasksJSON, nil)
+		if err != nil {
+			panic(err)
+		}
+	})
+
+	return comp
+}
+
+func (c *component) CreateExecution(sysVars map[string]any, setup *structpb.Struct, task string) (*base.ExecutionWrapper, error) {
+	ctx := context.Background()
+	githubClient := newClient(ctx, setup)
+	e := &execution{
+		ComponentExecution: base.ComponentExecution{Component: c, SystemVariables: sysVars, Setup: setup, Task: task},
+		client:             githubClient,
+	}
+
+	switch task {
+	case taskGetAllPRs:
+		e.execute = e.client.getAllPullRequestsTask
+	case taskGetPR:
+		e.execute = e.client.getPullRequestTask
+	case taskGetReviewComments:
+		e.execute = e.client.getAllReviewCommentsTask
+	case taskCreateReviewComment:
+		e.execute = e.client.createReviewCommentTask
+	case taskGetCommit:
+		e.execute = e.client.getCommitTask
+	case taskGetAllIssues:
+		e.execute = e.client.getAllIssuesTask
+	case taskGetIssue:
+		e.execute = e.client.getIssueTask
+	case taskCreateIssue:
+		e.execute = e.client.createIssueTask
+	case taskCreateWebhook:
+		e.execute = e.client.createWebhookTask
+	default:
+		return nil, errmsg.AddMessage(
+			fmt.Errorf("not supported task: %s", task),
+			fmt.Sprintf("%s task is not supported.", task),
+		)
+	}
+
+	return &base.ExecutionWrapper{Execution: e}, nil
+}
+
+func (e *execution) fillInDefaultValues(input *structpb.Struct) (*structpb.Struct, error) {
+	task := e.Task
+	// Fill in the default values for missing input
+	taskSpec, ok := e.Component.GetTaskInputSchemas()[task]
+	if !ok {
+		return nil, errmsg.AddMessage(
+			fmt.Errorf("task %s not found", task),
+			fmt.Sprintf("Task %s not found", task),
+		)
+	}
+	// Unmarshal the taskSpec into a map
+	var taskSpecMap map[string]interface{}
+	err := json.Unmarshal([]byte(taskSpec), &taskSpecMap)
+	if err != nil {
+		return nil, errmsg.AddMessage(
+			err,
+			"Failed to unmarshal input",
+		)
+	}
+	inputMap := taskSpecMap["properties"].(map[string]interface{})
+	// iterate over the inputMap and fill in the default values
+	for key, value := range inputMap {
+		valueMap, ok := value.(map[string]interface{})
+		if !ok {
+			// fmt.Println("value is not a map", key)
```

Comment:
`// fmt.Println("value is not a map", key)`
Please take them out.

Reply:
NIT
About memo, my preference is that we should make the code readable itself rather than using memo to explain code.

The memo is for the special decision or case.
So, I will not add those memo in the codebase.

`// iterate over the inputMap and fill in the default values`